### PR TITLE
Rendering refactor to support clipping and improve performance

### LINKED ISSUE #27: Spine clipping mask problem
I created animation in spine with clipping mask. After import to playcanvas these files and tried to run animation, I’m getting this error;

TypeError: Cannot read property ‘texture’ of undefined
at t.initAttachment (http://launch.playcanvas.com/api/assets/files/Scripts/Library/playcanvas-spine.min.js?id=27919451&branchId=a171ac64-c8db-441f-8ca3-219148ce95ba:3:147858 1)
at t.updateSlot (http://launch.playcanvas.com/api/assets/files/Scripts/Library/playcanvas-spine.min.js?id=27919451&branchId=a171ac64-c8db-441f-8ca3-219148ce95ba:3:152063)
at t.update (http://launch.playcanvas.com/api/assets/files/Scripts/Library/playcanvas-spine.min.js?id=27919451&branchId=a171ac64-c8db-441f-8ca3-219148ce95ba:3:153211)
at t.onUpdate (http://launch.playcanvas.com/api/assets/files/Scripts/Library/playcanvas-spine.min.js?id=27919451&branchId=a171ac64-c8db-441f-8ca3-219148ce95ba:3:155360)
at Function._helper (https://code.playcanvas.com/playcanvas-stable.dbg.js:24546:14)
at Function.update (https://code.playcanvas.com/playcanvas-stable.dbg.js:24554:10 1)
at Application.update (https://code.playcanvas.com/playcanvas-stable.dbg.js:23988:24)
at Application.tick (https://code.playcanvas.com/playcanvas-stable.dbg.js:24343:11 1)
at Application.start (https://code.playcanvas.com/playcanvas-stable.dbg.js:23977:10)
at http://launch.playcanvas.com/editor/scene/js/launch.js:7250:25 1

Example project: 
https://playcanvas.com/project/665578/overview/spine_mask_sample

Any idea?

### PR DESCRIPTION
**Pull request by mvaligursky**
Fixes: #27

### CODE REVIEW

**Review by willeastcott — APPROVED** [review]

File: Gruntfile.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ module.exports = function (grunt) {
         'src/component/spine-component-system.js',
         'src/component/spine-component.js',
         'src/component/spine-component-data.js',
-        'src/mesh-plugin.js',
+        'src/spine-plugin.js',
```
Comment (willeastcott) [inline Gruntfile.js]: Remove trailing comma.

File: Gruntfile.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ module.exports = function (grunt) {
...
     ];
     const src36 = [
         'contrib/spine-ts/build/3.6/spine-core.js',
```
Comment (willeastcott) [inline Gruntfile.js:10]: Trailing comma - plus another blow

**Comment by willeastcott** [comment]
Did you run the linter on your PR? It doesn't run automatically on this repo.

**Review by willeastcott — CHANGES_REQUESTED** [review]

**Comment by mvaligursky** [comment]
fixed lint errors and merged

**Pull request merged**